pmm-app/src/pmm-settings/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {
+  FC,
+  useState
+} from 'react';
+import { Form as FinalForm, Field as FinalField, FormRenderProps } from 'react-final-form';
+import {
+  Button,
+  LinkButton,
+  useTheme,
+} from '@grafana/ui';
+import { showSuccessNotification, showErrorNotification } from 'shared/components/helpers';
+import { ButtonWithSpinner } from 'shared/components/Form';
+import { getStyles } from './SignUp.styles';
+import { Messages } from './SignUp.messages';
+import { InputFieldAdapter, CheckboxFieldAdapter } from './FieldAdapters/FieldAdapters';
+import { TERMS_OF_SERVICE_URL, PRIVACY_POLICY_URL, NOTIFICATION_SETTINGS_URL } from './SignUp.constants';
+import { required, validEmail, composeValidators } from './SignUp.validators';
+import { SignUpData, SignUpProps } from './types';
+import { LoggedIn } from './LoggedIn/LoggedIn';
+
+const signUp = async (signUpData: SignUpData) => {
+  // XXX: stub for the actual sign up API call
+  signUpData;
+  await new Promise((resolve) => setTimeout(resolve, 1500));
+};
+
+export const SignUp: FC<SignUpProps> = ({ userEmail }) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+  const [loggedInEmail, setLoggedInEmail] = useState(userEmail);
+
+  const checkboxLabel = (
```

Comment:
It's better to respect the component naming convention, i.e. CheckboLabel

Reply:
Even though it's internal to to the parent component.

Reply:
the reason I didn't follow that naming convention is because that's not actually a component, but a piece of JSX code. In this case it would be even better to convert it to a component and pass it as `label={CheckboxLabel}`.

Reply:
Done